### ISSUE TITLE
[bitnami/airflow] Stack trace/errors do not show up for DAG tasks

### ISSUE THREAD

**Issue by dontspamterry** [bitnami/charts#6733]
**Which chart**:
bitnami/airflow 10.2.1

**Describe the bug**
Unable to debug what is failing for DAG tasks. For my DAG, task pod logs are minimal and when a task fails, there are no error logs/stack traces present in the task logs. Additionally, tasks which completed successfully or failed do not have their logs shipped to S3. The S3 export of task logs works for a set of example DAGs I used to test the S3 integration, but when I switch over to our business DAG, the S3 export ceases to work.

**To Reproduce**
I really don't know how to come up with a scenario to reproduce this. This lack of logs, especially any stack trace thrown from unhandled exceptions, is becoming problematic as we cannot figure out why a task fails. We have set the following configuration on the worker pods in the hopes of receiving more logging info:

```
worker:
  image:
    debug: "true"
  extraEnvVars:
   - name: AIRFLOW__LOGGING__LOGGING_LEVEL
     value: "debug"
```

but we still have no clear indication of what is failing in our task.

Is there other configurations we can set to expose more logging? We feel stack traces from unhandled exceptions may be potentially swallowed up, hence, not appearing in task logs

**Expected behavior**
Task logs

**Version of Helm and Kubernetes**:

- Output of `helm version`:

```
version.BuildInfo{Version:"v3.4.1", GitCommit:"c4e74854886b2efe3321e185578e6db9be0a6e29", GitTreeState:"clean", GoVersion:"go1.14.11"}

```

- Output of `kubectl version`:

```
Client Version: version.Info{Major:"1", Minor:"19", GitVersion:"v1.19.4", GitCommit:"d360454c9bcd1634cf4cc52d1867af5491dc9c5f", GitTreeState:"clean", BuildDate:"2020-11-11T13:17:17Z", GoVersion:"go1.15.2", Compiler:"gc", Platform:"linux/amd64"}
Server Version: version.Info{Major:"1", Minor:"19+", GitVersion:"v1.19.6-eks-49a6c0", GitCommit:"49a6c0bf091506e7bafcdb1b142351b69363355a", GitTreeState:"clean", BuildDate:"2020-12-23T22:10:21Z", GoVersion:"go1.15.5", Compiler:"gc", Platform:"linux/amd64"}
```

**Additional context**
None

**Comment by Mauraza** [comment]
Hi @dontspamterry,

Could you try with these values? 
```yaml
  - name: AIRFLOW__LOGGING__ENCRYPT_S3_LOGS
    value: "True"
  - name: AIRFLOW__LOGGING__LOGGING_LEVEL
    value: "DEBUG"
  - name: AIRFLOW__LOGGING__FAB_LOGGING_LEVEL
    value: "DEBUG"
```
To see if the logs give more information.

**Comment by dontspamterry** [comment]
Hi @Mauraza ,

I updated the config in my issue. Instead of the web pod, I updated to reflect the worker pod. My updated config is now:
```
worker:
  image:
    debug: "true"
  extraEnvVars:
    - name: AIRFLOW__LOGGING__ENCRYPT_S3_LOGS
      value: "True"
    - name: AIRFLOW__LOGGING__LOGGING_LEVEL
      value: "DEBUG"
    - name: AIRFLOW__LOGGING__REMOTE_LOGGING
      value: "True"
    - name: AIRFLOW__LOGGING__REMOTE_BASE_LOG_FOLDER
      value: <S3 folder>
```
I did not add the FAB_LOGGING_LEVEL as that only applies to the web pod, which is not causing problems.

I also had the additional env var set on my scheduler:
```
scheduler:
  extraEnvVars:
    - name: AIRFLOW__KUBERNETES__DELETE_WORKER_PODS
      value: "False"
```
so the worker pods do not get deleted after running so I can at least view the logs. I re-ran my DAG and same behavior - logs do not tell me what is failing. I believe there is an unhandled exception occurring somewhere, but the stack trace is not getting propagated to the pod log to help us in debugging.

As I stated previously, our set of example DAGs run fine and logs get shipped to S3. Once we swapped out to use an actual DAG for our business, none of our task logs get shipped, so it seems like there's two issues going on here.

Any other suggestions on what we can try?

**Comment by Mauraza** [comment]
Hi @dontspamterry,

Could you tell us where are you running the chart and share the values that you are using?

**Comment by dontspamterry** [comment]
Hi @Mauraza ,

We are running picking up Helm chart airflow 10.2.1 from https://charts.bitnami.com/bitnami. Our values override is quite large, so is there any specific sections you'd want from me?

Second question - we have deployed the Helm chart to Kubernetes. Since our worker pod is configured to not be destroyed, I'm able to view the container log for a given pod. I just created an example DAG that intentionally fails and the failed task log is properly shipped to S3 containing both DEBUG level and stack trace for my unhandled exception. I noticed that the task log is not the same as the container log. Do you know where the task log resides in worker pod? I'm wondering if the logging configuration on our end is causing the issues we are observing.

Thanks!

**Comment by Mauraza** [comment]
Hi @dontspamterry,

> We are running picking up Helm chart airflow 10.2.1 from https://charts.bitnami.com/bitnami. Our values override is quite large, so is there any specific sections you'd want from me?

The idea was to try to reproduce your issue.

> Second question - we have deployed the Helm chart to Kubernetes. Since our worker pod is configured to not be destroyed, I'm able to view the container log for a given pod. I just created an example DAG that intentionally fails and the failed task log is properly shipped to S3 containing both DEBUG level and stack trace for my unhandled exception. I noticed that the task log is not the same as the container log. Do you know where the task log resides in worker pod? I'm wondering if the logging configuration on our end is causing the issues we are observing.

Maybe you change the default log output of airflow, could you check this? 
https://airflow.apache.org/docs/apache-airflow/stable/logging-monitoring/logging-tasks.html

**Comment by dontspamterry** [comment]
Thanks @Mauraza . I'll close this for now. After more testing with some simple DAGs, we concluded there is something in the logging configuration for our DAG code that is preventing task logs from being dropped in to the expected location for shipment to S3. Our test DAGs stripped out any logging configuration and had both success and failure scenarios. In both cases, we saw those task logs shipped to S3. We don't know what the exact issue is yet on our code side, and the side effect it has on the worker task logs, so we'll continue to experiment. Thanks for helping!
